firmware/src/touch.c

```diff
@@ -155,17 +155,26 @@ bool touch_sensor_ok(unsigned i)
 
 const uint16_t *touch_raw()
 {
-    static uint16_t readout[36];
-    uint16_t buf[36];
+    static uint16_t readout[36] = {0};
+    // Do not use readout as buffer directly, update readout with buffer when operation finishes
+    uint16_t buf[36] = {0};
 
     for (int i = 0; i < 3; i++) {
         sensor_ok[i] = mpr121_raw(MPR121_BASE_ADDR + i, buf + i * 12, 12);
     }
+    memcpy(readout, buf, sizeof(readout));
+
+    return readout;
+}
+
+const uint16_t *map_raw_to_zones(const uint16_t* raw) {
```

Comment:
Just a tiny thing. `const uint16_t *raw`

I'll merge after you fix it.

Reply:
Fixed in latest comment